test/test_compile_part_4.py

```diff
@@ -292,6 +292,22 @@ def test_compile_with_esp8266_bundled_libraries(run_command, data_dir, copy_sket
     assert "\n".join(expected_output) not in res.stdout
 
 
+def test_generate_compile_commands_json_with_esp32(run_command, data_dir, copy_sketch):
+    # https://github.com/arduino/arduino-cli/issues/1547
+    assert run_command(["update"])
+
+    # Update index with esp32 core and install it
+    url = "https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_index.json"
+    assert run_command(["core", "update-index", f"--additional-urls={url}"])
+    assert run_command(["core", "install", "esp32:esp32", f"--additional-urls={url}"])
```

Comment:
```suggestion
    assert run_command(["core", "install", "esp32:esp32@2.0.1", f"--additional-urls={url}"])
```
I think it is safest to pin to a version of the boards platform we know will produce the conditions required for an effective test.